### ISSUE TITLE
open file dialog not implemented for osx

### ISSUE THREAD

**Issue by toejough** [beeware/toga#646]
## Expected Behavior
I expected open_file_dialog to work for osx.

## Current Behavior
```
File "<venv>/lib/python3.7/site-packages/toga/window.py", line 275, in open_file_dialog
    return self._impl.open_file_dialog(title, initial_directory, file_types, multiselect)
AttributeError: 'MainWindow' object has no attribute 'open_file_dialog'
```

## Steps to reproduce
`app.main_window.open_file_dialog(...`

## Your Environment
* Python Version
3.7.2

* Operating System and Version (select from the following and list the specific version number; if your OS is not listed, list that as well)

  - [x] macOS - version: 10.14.3
  - [ ] Linux - distro:   - version:
  - [ ] Windows - version:
  - [ ] Other - name:  - version:

* Toga Target (the type of app you are trying to generate)
 
  - [ ] android
  - [x] cocoa
  - [ ] django 
  - [ ] gtk
  - [ ] iOS
  - [ ] tvOS
  - [ ] watchOS
  - [ ] winforms 
  - [ ] win32
  - [ ] Other (please specify)

**Comment by ghynes** [comment]
Are there any solutions or work arounds for this one?
I have that same issue as toejough with more or less the same environmental setup

**Comment by freakboy3742** [comment]
@ghynes Well the solution for something that isn't implemented is... someone implements it. 😄 

The workaround would be to dig into how the other file dialogs work on Cocoa, and access the same underlying APIs - however, once you've done that, you'd be 90% of the way to implementing a full solution...

**Comment by nehararora** [comment]
I've created a PR that fixes this - #674. I'd be happy to make changes as needed :)
